Unify make and cmake behaviours

TGWDB opened the pull request:
This fixes the inconsistencies between cmake and make
behaviours.

Fixes #5986 

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

TGWDB committed:
Unify make and cmake behaviours

This fixes the inconsistencies between cmake and make
behaviours.

tautschnig commented on regression/cbmc/Makefile:
```diff
@@ -11,6 +13,8 @@ endif
 
 test:
 	@../test.pl -e -p -c "../../../src/cbmc/cbmc --validate-goto-model --validate-ssa-equation" -X smt-backend $(GCC_ONLY)
+	@$(MAKE) -f $(THIS_FILE) test-paths-lifo
+	@$(MAKE) -f $(THIS_FILE) test-cprover-smt2
```
How about making the `test` target depend on `test-paths-lifo` and `test-cprover-smt2`?

SaswatPadhi replied:
Yes, either that, or make `default` target depend on `test-paths-lifo` and `test-cprover-smt2`?

SaswatPadhi replied:
One issue to consider here is: if we add these additional targets as dependencies for `test`, we wouldn't be able to run "just `test`" ...

tautschnig replied:
Good call. The right approach probably is to rename the existing `test` target and introduce a new (`PHONY`) `test` target.

TGWDB committed:
Use dependency style make

Change from calling the separate make targets inside a default
to instead using make dependency style.

tautschnig commented:
I'm wondering whether tests would pass if this were rebased onto current develop, which includes #5985.

TGWDB committed:
Unify the other way

tautschnig commented on regression/Makefile:
```diff
@@ -72,14 +72,14 @@ endif
 $(DIRS):
 	@echo "Running $@..." ;
-	$(MAKE) -C "$@" || exit 1;
+	$(MAKE) -C "$@" test || exit 1;
```
Wouldn't `test-parallel` (below) also need this change?
